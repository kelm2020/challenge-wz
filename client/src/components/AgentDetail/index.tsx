import React, { useState, useEffect, useCallback, FC } from 'react';
import Card from '@mui/material/Card';
import Button from '@mui/material/Button';
import CardActions from '@mui/material/CardActions';
import { CircularProgress } from '@material-ui/core';
import { useParams } from 'react-router-dom';
import { DataAgent, IDetail } from '../../interfaces'
import { usePrevious } from '../../utils';

const style = {
  width: '100%',
  maxWidth: '800px',
  maxHeight: '800px',
  border: '2px solid #000',
  borderRadius: '8px',
  marginLeft: 'auto',
  marginRight: 'auto'
};

const styleTextArea = { width: '100%', height: '400px' };

const AgentModal: FC<IDetail> = ({eventHandler}: IDetail) => {
  const [ dataAgent, setDataAgent ] = useState<DataAgent | undefined>(undefined);

  const params = useParams();
  const { id } = params;
  const get = useCallback(async () => {
    try {
      const response = await fetch(`http://localhost:5000/agents/${id}`)
      const jsonResponse = await response.json()
      setDataAgent(jsonResponse.data)
    } catch (e) {
      console.error(e);
    }
  }, [id]);

  const prevId = usePrevious(id);

  useEffect(() => {
    if(prevId !== id) {
      get()
    }
  }, [id, prevId]);

  return (
    <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', margin: '20px' }}>
      <Card style={style} sx={{ minWidth: 275 }}>
        <>
          {
            dataAgent === undefined ? <CircularProgress /> :
            <textarea style={styleTextArea}>
              {JSON.stringify(dataAgent, undefined, 4)}
            </textarea>
          }
        </>
        <CardActions>
          <Button onClick={eventHandler} size="small">Atras</Button>
        </CardActions>
      </Card>
    </div>
  );
};

export default AgentModal;