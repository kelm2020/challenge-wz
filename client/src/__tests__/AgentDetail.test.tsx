import { render, fireEvent, screen } from '@testing-library/react';
import AgentDetail from '../components/AgentDetail';

describe('Tests of Agent Detail', () => {
  it('should match snapshot', () => {
    const props = {
      eventHandler: jest.fn(),
    };

    const { asFragment } = render(<AgentDetail {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test get back event handler', async () => {
    const props = {
      eventHandler: jest.fn(),
    };

    render(<AgentDetail {...props} />);
    fireEvent.click(screen.getByRole('button'));
    expect(props.eventHandler).toBeCalled();
  });

});