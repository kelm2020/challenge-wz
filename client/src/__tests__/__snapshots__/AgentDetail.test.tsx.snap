// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of Agent Detail should match snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: flex; justify-content: center; align-items: center; margin: 20px;"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-pwngrv-MuiPaper-root-MuiCard-root"
      style="width: 100%; max-width: 800px; max-height: 800px; border: 2px solid #000; border-radius: 8px; margin-left: auto; margin-right: auto;"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Atras
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
